File: src/net/eduard/economy/command/EconomyTopCommand.java

package net.eduard.economy.command;

import java.util.List;
import java.util.Map;
import java.util.Map.Entry;
import java.util.stream.Collectors;
import java.util.stream.Stream;

import net.eduard.economy.EduEconomy;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.scheduler.BukkitRunnable;

import net.eduard.api.lib.manager.CommandManager;
import net.eduard.api.lib.modules.Extra;
import net.eduard.api.lib.game.FakePlayer;

public class EconomyTopCommand extends CommandManager {

    public EconomyTopCommand() {
        super("top", "rank");
        setDescription("Verificar o top de dinheiro");
    }

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        show(sender);
        return true;
    }

    public void show(CommandSender sender) {
        Map<FakePlayer, Double> currency = EduEconomy.getInstance().getManager().getCurrency();
        Stream<Entry<FakePlayer, Double>> streamOrdenada = currency.entrySet().stream()
                .sorted((entry1, entry2) -> entry2.getValue().compareTo(entry1.getValue()))
                .limit(EduEconomy.getInstance().getMessages().getInt("top-size"));
        List<Entry<FakePlayer, Double>> lista = streamOrdenada.collect(Collectors.toList());
        sender.sendMessage(EduEconomy.getInstance().getMessages().message("top-format-header"));
        int posicao = 1;
        for (Entry<FakePlayer, Double> entrada : lista) {
            sender.sendMessage(EduEconomy.getInstance().getMessages().message("top-format")
                    .replace("$player", entrada.getKey().getName())
                    .replace("$amount", Extra.formatMoney(entrada.getValue())).replace("$position", "" + posicao));
            posicao++;
        }

    }

}

File: src/main/java/net/eduard/economy/command/EconomyTopCommand.kt
package net.eduard.economy.command

import net.eduard.api.lib.kotlin.fake
import net.eduard.api.lib.manager.CommandManager
import net.eduard.api.lib.modules.Extra
import net.eduard.economy.EduEconomyPlugin
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class EconomyTopCommand : CommandManager("top", "rank") {
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {

        val instance = EduEconomyPlugin.instance
        sender.sendMessage(instance.messages.message("top-format-header"))
        for ((user, money) in instance.manager.top) {
            val playerName = user.name
            val pos = user.lastTopPosition
            sender.sendMessage(instance.messages.message("top-format")
                .replace("%player", playerName)
                .replace("%amount", Extra.formatMoney(money))
                .replace("%position", "" + pos)
            )
        }
        if (sender is Player){
            val user = instance.manager.getAccount(sender.fake)
            sender.sendMessage("")
            sender.sendMessage("§6Sua posição é §e${user.lastTopPosition}º §6atualmente.")
        }

        return true
    }

    init {
        description = "Verificar o top de dinheiro"
    }
}